src/Data/List/Base.agda

```diff
@@ -252,6 +250,11 @@ tail : List A → Maybe (List A)
 tail []       = nothing
 tail (_ ∷ xs) = just xs
 
+last : List A → Maybe A
+last []           = nothing
+last (x ∷ [])     = just x
+last (x ∷ y ∷ xs) = last (y ∷ xs)
```

Comment:
```suggestion
last (x ∷ xs) = last xs
```

Reply:
Out of interest, are there any advantages to this aside from it being shorter?

Reply:
It may save a tiny bit by not having to reallocate a cons
node that is virtually equal to the one taken apart but it's
mostly a readability concern tbh.